tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -646,6 +687,217 @@ HRESULT STDMETHODCALLTYPE CorProfiler::ModuleLoadFinished(ModuleID module_id, HR
+            auto enumTypeRefs = Enumerator<mdTypeRef>(
+                [&metadata_import](HCORENUM* ptr, mdTypeRef arr[], ULONG max, ULONG* cnt) -> HRESULT {
+                    return metadata_import->EnumTypeRefs(ptr, arr, max, cnt);
+                },
+                [&metadata_import](HCORENUM ptr) -> void { metadata_import->CloseEnum(ptr); });
+
+            auto enumIterator = enumTypeRefs.begin();
+            while (enumIterator != enumTypeRefs.end())
+            {
+                mdTypeRef typeRef = *enumIterator;
+
+                // Check if the typeref matches
+                mdToken parent_token = mdTokenNil;
+                WCHAR type_name[kNameMaxSize]{};
+                DWORD type_name_len = 0;
+
+                hr = metadata_import->GetTypeRefProps(typeRef, &parent_token, type_name, kNameMaxSize,
+                                                      &type_name_len);
+
+                if (TypeNameMatchesTraceAttribute(type_name, type_name_len))
+                {
+                    foundType = true;
+                    Logger::Debug("ModuleLoadFinished found the TypeRef for ", traceAttribute,
+                                 " defined in Module ", module_info.assembly.name);
+                    break;
+                }
+
+                enumIterator = ++enumIterator;
+            }
+        }
+
+        // We have typeRef and it matches Datadog.Trace.Attribute
+        // Since it is referenced, it should be in-use somewhere in this module
+        // So iterate over all methods in the module
+        if (foundType)
+        {
+            std::vector<MethodReference> methodReferences;
+            std::vector<IntegrationDefinition> integrationDefinitions;
+
+            // Now we enumerate all custom attributes in this assembly to see if Datadog.Trace.Annotations.TraceAttribute is
+            // used
+            auto enumCustomAttributes = Enumerator<mdCustomAttribute>(
+                [&metadata_import](HCORENUM* ptr, mdCustomAttribute arr[], ULONG max, ULONG* cnt) -> HRESULT {
+                    return metadata_import->EnumCustomAttributes(ptr, mdTokenNil, mdTokenNil, arr, max, cnt);
+                },
+                [&metadata_import](HCORENUM ptr) -> void { metadata_import->CloseEnum(ptr); });
+            auto customAttributesIterator = enumCustomAttributes.begin();
+
+            while (customAttributesIterator != enumCustomAttributes.end())
+            {
+                mdCustomAttribute customAttribute = *customAttributesIterator;
+
+                // Check if the typeref matches
+                mdToken parent_token = mdTokenNil;
+                mdToken attribute_ctor_token = mdTokenNil;
+                const void* attribute_data = nullptr; // We'll likely need to use this Hopefully we don't need to
+                                                        // use this because I don't know how
```

Comment:
conflicted! 😄

Reply:
😆 I'll clean this up now that it's implemented 😛

Reply:
🤣🤣🤣

Reply:
Existential crisis fixed in 46ae5f8dd067fd13db2b590f9a283f7ed6c1edea